Vester/Private/Template/VesterTemplate.Tests.ps1

```diff
@@ -22,18 +22,50 @@ Param(
     [switch]$Remediate
 )
 
-# Process .Vester.ps1 files one at a time
-ForEach ($Test in $TestFiles) {
-    Write-Verbose "Processing test file $Test"
-    $TestName = Split-Path $Test -Leaf
-    $Scope = Split-Path (Split-Path $Test -Parent) -Leaf
+# Gets the scope, the objects for the scope and their requested test files
+$Scopes = Split-Path (Split-Path $TestFiles -Parent) -Leaf | Select -Unique
+$Final = @()
+$InventoryList = @()
+$Datacenter = Get-Datacenter -Name $cfg.scope.datacenter -Server $cfg.vcenter.vc
+foreach($Scope in $Scopes)
+{
+    Write-Verbose "Processing $Scope"
+    Remove-Variable InventoryList -ErrorAction SilentlyContinue # Makes sure the variable is always fresh
+    # Use $Scope (parent folder) to get the correct objects to test against
+    # If changing values here, update the "$Scope -notmatch" test below as well
+    $InventoryList = switch ($Scope) {
+        'vCenter'    {$cfg.vcenter.vc}
+        'Datacenter' {$Datacenter}
+        'Cluster'    {$Datacenter | Get-Cluster -Name $cfg.scope.cluster}
+        'DSCluster'  {$Datacenter | Get-DatastoreCluster -Name $cfg.scope.dscluster}
+        'Host'       {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VMHost -Name $cfg.scope.host}
```

Comment:
Is it possible to use $Cluster | Get-VMHost instead of getting the cluster a second time?  Not a huge deal, and we can leave this code alone for this PR if necessary.